create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/service/IdentityUpdater.java

```diff
@@ -71,7 +72,9 @@ private static Identity updateIdentity(BareProxyClient bareProxyClient, Identity
             throws URISyntaxException {
         try {
             Optional<String> arpId = bareProxyClient.lookupArpidForOrcid(identity.getOrcId());
-            arpId.ifPresent(identity::setArpId);
+            if (arpId.isPresent()) {
+                identity.setId(URI.create(arpId.get()));
```

Comment:
What happens if both ORCID and ARP-ID finnes?

Reply:
The ARP-ID from ARP found by looking up by  ORCID will overwrite eventually existing id in the Identity, but the ArpID/ORCID combination is supposed to have been confirmed in ARP.